src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -101,28 +111,38 @@ data CCurrency
     deriving (Show, Read, Generic)
 
 -- | Client hash
-newtype CHash = CHash Text deriving (Show, Eq, Generic, Buildable)
+newtype CHash = CHash Text
+    deriving (Show, Eq, Generic, Buildable)
 
 instance Hashable CHash where
     hashWithSalt s (CHash h) = hashWithSalt s h
 
 -- | Client address
-newtype CAddress = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+-- @w@ is phantom type and stands for type of item this address belongs to.
+newtype CAddress w = CAddress CHash
```

Comment:
I used phantom type, because in all cases `CAddress` is created in the same way (with same constructor); but I want type of account address to differ from type of walletset address, not to confuse them in use (they have different semantics).

> If we are using phantom types, we could at least restrict the types to just the two cases

@ksaric Could you also expand on this?